internal/service/instance.go

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) F5, Inc.
+ *
+ * This source code is licensed under the Apache License, Version 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package service
```

Comment:
Is this the package doing the heavy lifting? Is the plan for this Service layer to be the brains of the Agent? Seems like the plugin layer is in this implementation?

Reply:
Yes the service layer should brains of the agent. At the moment its not doing much since its just getting NGINX instance data. 
The plugin layer will delegate requests to the service layer which will determine what to do with the requests and call out to the appropriate datasource.